compiler/luci/lang/include/luci/IR/SparsityParam.h

```diff
@@ -37,23 +39,186 @@ enum SparseIndexVectorType
   U8,
 };
 
-struct DimMetaData
+class SparseIndexVector
 {
+public:
+  SparseIndexVector() = default;
```

Comment:
I think default CTOR should be removed
```cxx
SparseIndexVector() = delete;
```
Current DTOR uses `_type` to deletes what was created

Reply:
Maybe remove copy CTOR too?

Reply:
```cxx
DimMetaData(DimensionType format, int32_t dense_size) : _format{format}, _dense_size{dense_size}
  {
    // DO NOTHING
  }
```
`DimMetaData` can be two types, e.g., dense, sparse. when it is sparse, `DimMetaData` calls above CTOR that calls default `SparseIndexVector` CTOR.

Reply:
Ah... didn't read the code carefully :)